Docs update (#1965)
graphql-dotnet/graphql-dotnet: README.md

# GraphQL for .NET

[![Join the chat at https://gitter.im/graphql-dotnet/graphql-dotnet](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/graphql-dotnet/graphql-dotnet?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)

> [![Build status](https://github.com/graphql-dotnet/graphql-dotnet/workflows/Build%20artifacts/badge.svg)](https://github.com/graphql-dotnet/graphql-dotnet/actions)

Author: I wanted to give an exact link  like https://github.com/graphql-dotnet/graphql-dotnet/actions?query=workflow%3A%22Publish+code%22, but there were quotes in it and the markdown refused.

Reviewer: Try without the quotes?

https://github.com/graphql-dotnet/graphql-dotnet/actions?query=workflow%3APublish+code

I tested it and it seems to work.....

Reviewer: ```suggestion
[![Build status](https://github.com/graphql-dotnet/graphql-dotnet/workflows/Build%20artifacts/badge.svg)](https://github.com/graphql-dotnet/graphql-dotnet/actions?query=workflow%3ABuild+artifacts)
```

Author: I also thought so at first, but take a closer look, the search string is truncated after the first space.
![изображение](https://user-images.githubusercontent.com/21261007/97502775-e1d2d400-1984-11eb-9275-307dabe9c298.png)

Reviewer: You're right.  Oh well.

> [![Build status](https://github.com/graphql-dotnet/graphql-dotnet/workflows/Publish%20code/badge.svg)](https://github.com/graphql-dotnet/graphql-dotnet/actions)

Reviewer: ```suggestion
[![Build status](https://github.com/graphql-dotnet/graphql-dotnet/workflows/Publish%20code/badge.svg)](https://github.com/graphql-dotnet/graphql-dotnet/actions?query=workflow%3APublish+code)
```

> `AllowSynchronousIO` to `true` as per [this announcement](https://github.com/aspnet/Announcements/issues/342);
> which isn't recommended.*

> You can get all preview versions from [GitHub registry](https://github.com/orgs/graphql-dotnet/packages?repo_name=graphql-dotnet).

Reviewer: ```suggestion
You can get all preview versions from [GitHub Packages](https://github.com/orgs/graphql-dotnet/packages?repo_name=graphql-dotnet).
Note that GitHub requires authentication to consume the feed.  See [here](https://docs.github.com/en/free-pro-team@latest/packages/publishing-and-managing-packages/about-github-packages#authenticating-to-github-packages).
```

Reviewer: It would be nice to have a little more documentation about how to actually use a package from GitHub Packages, with a sample NuGet.config file and an outline of how to create a PAT and configure a Windows computer to authenticate and use the feed.  Perhaps inside the docs rather than here in the readme, with a link to the docs from here.

Author: You understand this more than I do.

GitHub Packages or GitHub registry ? First I wrote Packages but then switched to registry. How is it correct?

Reviewer: GitHub Packages is the name of GitHub’s registry

Reviewer: Since all their documentation uses “GitHub Packages” (as it relates to this), and the packages are listed under the title of “Packages” on our repo homepage, I’d use “GitHub Packages”

Reviewer: Maybe I can add some docs....

  - [x] directives

### Publishing NuGet packages

The package publishing process is automated with [GitHub Actions](https://github.com/features/actions).

> After your PR is merged into `master` or `develop` preview packages are published to [GitHub registry](https://github.com/orgs/graphql-dotnet/packages?repo_name=graphql-dotnet).

Reviewer: ```suggestion
After your PR is merged into `master` or `develop`, preview packages are published to [GitHub Packages](https://github.com/orgs/graphql-dotnet/packages?repo_name=graphql-dotnet).
```

> Stable versions of packages are published to Nuget when a [release](https://github.com/graphql-dotnet/graphql-dotnet/releases) is created.

Reviewer: ```suggestion
Stable versions of packages are published to NuGet when a [release](https://github.com/graphql-dotnet/graphql-dotnet/releases) is created.
```
